packages/core/data-transfer/lib/providers/remote-strapi-destination-provider/index.ts

```diff
@@ -44,61 +45,28 @@ class RemoteStrapiDestinationProvider implements IDestinationProvider {
 
   type: ProviderType = 'destination';
 
-  options: IRemoteStrapiDestinationProvider;
+  options: IRemoteStrapiDestinationProviderOptions;
 
   ws: WebSocket | null;
 
-  constructor(options: IRemoteStrapiDestinationProvider) {
+  constructor(options: IRemoteStrapiDestinationProviderOptions) {
     this.options = options;
     this.ws = null;
   }
 
-  async #dispatch<U = unknown, T extends object = object>(message: T): Promise<U> {
-    const { ws } = this;
-
-    if (!ws) {
-      throw new Error('No ws connection found');
-    }
-
-    return new Promise((resolve, reject) => {
-      const uuid = v4();
-      const payload = JSON.stringify({ ...message, uuid });
-
-      ws.send(payload, (error) => {
-        if (error) {
-          reject(error);
-        }
-      });
-
-      ws.once('message', (raw) => {
-        const response: { uuid: string; data: U; error: string | null } = JSON.parse(
```

Comment:
JSON.parse is prone to throwing errors, do we want to wrap this in a try/catch and handle it on our own terms, otherwise I think this will get an "error thrown in an async method" if it happens.